sn_networking/src/metrics/bad_node.rs

```diff
@@ -168,18 +166,14 @@ impl BadNodeMetrics {
 impl ShunnedByCloseGroup {
     pub(crate) fn record_shunned_metric(&mut self, peer: PeerId) {
         // increment the metric if the peer is in the close group (new or old) and hasn't shunned us before
-        if self.close_group_peers.contains(&peer) {
-            if !self.close_group_peers_that_have_shunned_us.contains(&peer) {
+        if !self.old_new_group_shunned_list.contains(&peer) {
+            if self.close_group_peers.contains(&peer) {
                 self.metric_current_group.inc();
-                self.close_group_peers_that_have_shunned_us.insert(peer);
+                self.old_new_group_shunned_list.insert(peer);
```

Comment:
>this can be outside of the inner if clause but within the outer if clause
so that only one code line of insertion will be enough.

Do you mean within this clause? ` !self.old_new_group_shunned_list.contains(&peer)`
But that would contain ALL peers that are observed by us. I only want the ones that are in the close group (either old / new).